spec/app/domain/authentication/authn-oidc/update_input_with_username_from_id_token_spec.rb

```diff
@@ -42,23 +45,35 @@ def request_body(request)
   end
 
   let(:authenticate_id_token_request) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"alice\"}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"alice\"}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_missing_id_token_username_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_empty_id_token_username_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"\"}")
+    mock_authenticate_oidc_request(request_body_data: "id_token={\"id_token_username_field\": \"\"}", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_missing_id_token_field) do
-    mock_authenticate_oidc_request(request_body_data: "some_key=some_value")
+    mock_authenticate_oidc_request(request_body_data: "some_key=some_value", request_headers: {"CONTENT_TYPE" => "text/plain"})
   end
 
   let(:authenticate_id_token_request_empty_id_token_field) do
-    mock_authenticate_oidc_request(request_body_data: "id_token=")
+    mock_authenticate_oidc_request(request_body_data: "id_token=", request_headers: {"CONTENT_TYPE" => "text/plain"})
+  end
+
+  let(:authenticate_id_token_request_id_token_in_header_field) do
+    mock_authenticate_oidc_request(request_body_data: "id_token=", request_headers: {"HTTP_AUTHORIZATION" => "Bearer {\"id_token_username_field\":\"alice\"}"})
+  end
+
+  let(:authenticate_id_token_request_invalid_id_token_in_header_field) do
+    mock_authenticate_oidc_request(request_body_data: "id_token=", request_headers: {"HTTP_AUTHORIZATION" => "{\"id_token_username_field\":\"alice\"}"})
```

Comment:
In general, I see that the given token is expected to be in cleartext and not in base64. Will that work for our new flow we need to support?

Reply:
yes - they both work - I have checked